jobs/jobs_utils.py

```diff
@@ -0,0 +1,44 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper functions for beam validators and one-off jobs. The clone function
+should be used in every beam function since beam functions require all input
+to be immutable."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+
+def clone_model(model, **new_values):
+    """Clones the entity, adding or overriding constructor attributes.
+
+    The cloned entity will have exactly the same property values as the
+    original entity, except where overridden. By default, it will have no
+    parent entity or key name, unless supplied.
+
+    Args:
+        model: datastore_services.Model. Model to clone.
+        **new_values: dict(str: *). Keyword arguments to override when
+            invoking the cloned entity's constructor.
+
+    Returns:
+        *. A cloned, and possibly modified, copy of self. Subclasses of
```

Comment:
Instead of *, use the "lowest common ancestor" class name. E.g. datastore_services.Model I guess?

Reply:
Done.